bpf/lib/nodeport.h

```diff
@@ -1631,14 +1633,16 @@ int tail_nodeport_nat_ingress_ipv4(struct __ctx_buff *ctx)
 
 	ret = snat_v4_rev_nat(ctx, &target);
 	if (IS_ERR(ret)) {
-		/* In case of no mapping, recircle back to main path. SNAT is very
-		 * expensive in terms of instructions (since we don't have BPF to
-		 * BPF calls as we use tail calls) and complexity, hence this is
-		 * done inside a tail call here.
-		 */
-		ctx_skip_nodeport_set(ctx);
-		ep_tail_call(ctx, CILIUM_CALL_IPV4_FROM_NETDEV);
-		ret = DROP_MISSED_TAIL_CALL;
+		if (ret == DROP_NAT_NOT_NEEDED || ret == DROP_NAT_NO_MAPPING) {
```

Comment:
The NAT code seems to consistently use `NAT_PUNT_TO_STACK` as alias for `DROP_NAT_NOT_NEEDED`. We should be doing the same here (for both IPv4 and IPv6).

Reply:
Yes that makes sense, I can perhaps provide a change that completely remove definition of `DROP_NAT_NOT_NEEDED`, sounds good?